services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java

```diff
@@ -120,22 +125,29 @@ private static S3AsyncClient initializeS3AsyncClient(DefaultS3CrtClientBuilder b
-                                                                 .checksumValidationEnabled(false)
-                                                                 .build())
-                            .region(builder.region)
-                            .endpointOverride(builder.endpointOverride)
-                            .credentialsProvider(builder.credentialsProvider)
-                            .overrideConfiguration(overrideConfigurationBuilder.build())
-                            .accelerate(builder.accelerate)
-                            .forcePathStyle(builder.forcePathStyle)
-                            .crossRegionAccessEnabled(builder.crossRegionAccessEnabled)
-                            .putAuthScheme(new CrtS3ExpressNoOpAuthScheme())
-                            .httpClientBuilder(initializeS3CrtAsyncHttpClient(builder))
-                            .build();
+        S3AsyncClientBuilder s3AsyncClientBuilder = S3AsyncClient.builder()
+                                                                 // Disable checksum for streaming operations, it is handled in
+                                                                 // CRT. Checksumming for non-streaming
+                                                                 // operations is still handled in HttpChecksumStage
+                                                                 .serviceConfiguration(S3Configuration.builder()
+                                                                                                      .checksumValidationEnabled(false)
+                                                                                                      .build())
+                                                                 .region(builder.region)
+                                                                 .endpointOverride(builder.endpointOverride)
+                                                                 .credentialsProvider(builder.credentialsProvider)
+                                                                 .overrideConfiguration(overrideConfigurationBuilder.build())
+                                                                 .accelerate(builder.accelerate)
+                                                                 .forcePathStyle(builder.forcePathStyle)
+                                                                 .crossRegionAccessEnabled(builder.crossRegionAccessEnabled)
+                                                                 .putAuthScheme(new CrtS3ExpressNoOpAuthScheme())
+                                                                 .httpClientBuilder(initializeS3CrtAsyncHttpClient(builder));
+
+
+        if (builder.futureCompletionExecutor != null) {
+            s3AsyncClientBuilder.asyncConfiguration(b -> b.advancedOption(FUTURE_COMPLETION_EXECUTOR,
+                                                                          builder.futureCompletionExecutor).build());
```

Comment:
No need to call build on the `ClientAsyncConfiguration.Builder` in the lambda here

Reply:
Good catch, fixed